### DISCUSSION TITLE
Why use yup library ?

### DISCUSSION THREAD

**Discussion by hexionaut** [react-hook-form/react-hook-form#2486] [General]
Hi there,

I see a lot of people using `yup` library to validate inputs.

My question is, why ? I am currently discovering `react-hook-form` but I can't understand what `yup` does that `react-hook-form` can't do ?

Thanks !

**Comment by bluebill1049** [comment]
honestly, it's perfectly fine without it... and you save those extra package size as well...

**Reply by GabrielGomesBarbosa** [reply]
Exactly, i never got problem with the `react-hook-form` validation! its woking very good without any lib for validation!
